File: comparison/key.r
library(tidyverse)

sweaters <- read_csv("ugly_sweaters.csv") %>% 
  dplyr::filter(hs_tf == "Yes")

tidy_colors <- sweaters %>% 
  tidyr::separate_rows(colors, sep = ", ") %>% 
  tidyr::separate_rows(colors, sep = " and ") %>% 
  dplyr::group_by(sweater) %>% 
  dplyr::summarize(num_colors = length(sweater))

File: comparison/key.py
import pandas as pd

raw_data = pd.read_csv('ugly_sweaters.csv', index_col='sweater')
sweaters = raw_data[raw_data['hs_tf'] == 'Yes']

sweaters['colors_list'] = sweaters['colors'].str.split(',')
color_data = sweaters.explode('colors_list')
sweaters['num_colors'] = color_data.groupby('sweater')['colors_list'].count()